scripts/.common.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+# .common.sh by Dustyn Gibson <miigotu@gmail.com>
+
+#DEBUG=1
+
+BINUTILS="binutils-2.22"
+GCC="gcc-PS3"
+GDB="gdb-7.5.1"
+NEWLIB="newlib-1.20.0"
+
+TOOLCHAIN_ROOT=$(git rev-parse --show-toplevel)
```

Comment:
Probably better to use `realpath dirname $0/..` here.  We might not have the `.git` directory if this is from e.g. a `git archive`.

Reply:
A tarball or zip downloaded will not have the code for submodules in it, there must be some way around that but honestly nobody should need to build the toolchain anymore after I get the builds completed. running the docker image will give a mounted volume at /usr/local/ps3dev, and i plan to also upload the artifacts as releases (zipped ps3dev folder) for each arch.

Really the only people building it will be developers hacking on the toolchain anymore since every change/update will automatically be built and become readily available.

Reply:
And thats a mounted volume on the host* btw, you dont need to enter the docker shell.

Reply:
Why won't the tarball have the code for submodules in it?  How do you get the code for the submodules then?  If that doesn't work then it sounds like submodules still has at least one major flaw...

Are you going to publish Dockers for ppc64 and aarch64?  Then at least I don't need to do it anymore.  :laughing:
Artifacts for for SPARC, MIPS and RV64 (not supported by Docker at this juncture, AFAIK)?

And never underestimate the number of people who want to build the sw themselves just so they can feel sure there is no funny business going on.

Reply:
Dockers will be pushed for amd64/x86_64, i386, arm7, ppc64, s390x, etc. Basically anything you can run linux on I can push docker images for

Reply:
Thats what makes the github action yml so busy lol, multiarch

Reply:
> And never underestimate the number of people who want to build the sw themselves just so they can feel sure there is no funny business going on.

oh, yes, I'm one of those guys. I enjoy having my own "home-made" toolchain. 😄 
Also sometimes you want to build the toolchain for an OS/arch that might not be supported and then a manual build is the only option.

From my limited experience, I'd say that ready-to-use builds are great way to make new people jump in and try to "get a taste" of console development. Actually, I've seen a lot of people still using Estwald's PSDK3v2-build because the MinGW binaries were there to download & run.

_off-topic_, @zeldin , thanks for all the work you did on the SEGA DC 🌀 . (you can't imagine how many days/nights I spent doing Dreamcast development when I was in college) 😄

Reply:
The toolchain will always be able to be built locally, only it requires a git clone currently rather than a tarball until I figure out how to have the source of the submodules included in the zip/tar

Reply:
@miigotu Unfortunately there's plenty of things you can run linux on which you can't make Docker images for, because Docker is written in `go` so it has rather limited arch support...  (Unless it's possible to build Docker with `gccgo`, that might me interresting to try some time.)